UnitTests/CacheTest/MemoryCacheTest/UserFacebookProfilePictureCacheTest.cs

```diff
@@ -0,0 +1,304 @@
+                var imageBytes = webClient.DownloadData("http://graph.facebook.com/" + facebookId + "/picture?type=large");
+                if (imageBytes != null && imageBytes.Length > 0)
+                {
+                    imageBitmap = BitmapFactory.DecodeByteArray(imageBytes, 0, imageBytes.Length);
+                }
+            }
+            
+            // Add it to the cache
+            UserFacebookProfilePictureCache userFbPicCache = UserFacebookProfilePictureCache.GetInstance();
+            userFbPicCache.AddUserFacebookProfilePicture(11111, imageBitmap);
+
+            // Ensure that it is present in the cache and correctly associated with the facebook id
+            Dictionary<long, Bitmap> cache = GetObjectField<Dictionary<long, Bitmap>>(userFbPicCache, "lruCache");
+            Bitmap cacheImageBitmap;
+            cache.TryGetValue(facebookId, out cacheImageBitmap);
+            Assert.True(cacheImageBitmap != null && cacheImageBitmap.Equals(imageBitmap));
+
+            // Ensure that the facebook id is also present on the list of recently accessed
+            List<long> listRecentlyAccessed = GetObjectField<List<long>>(userFbPicCache, "listUserFacebookIdRecentlyAccessed");
+            Assert.True(listRecentlyAccessed.Contains(facebookId));
+        }
+
+        /// <summary>
+        /// Basic scenario in which we ensure that we are able to retrieve back an item from the cache, given
+        /// that it is present.
+        /// </summary>
+        [Test]
+        public void FindUserFacebookProfilePicture()
+        {
+            long facebookId = 11111;
+
+            // Get the profile picture
+            Bitmap imageBitmap = null;
+            using (var webClient = new WebClient())
+            {
+                var imageBytes = webClient.DownloadData("http://graph.facebook.com/" + facebookId + "/picture?type=large");
+                if (imageBytes != null && imageBytes.Length > 0)
+                {
+                    imageBitmap = BitmapFactory.DecodeByteArray(imageBytes, 0, imageBytes.Length);
+                }
+            }
+
+            // Add it to the cache using reflection (this is to make this test independent of the add method)
+            UserFacebookProfilePictureCache userFbPicCache = UserFacebookProfilePictureCache.GetInstance();
+            Dictionary<long, Bitmap> cache = GetObjectField<Dictionary<long, Bitmap>>(userFbPicCache, "lruCache");
+            cache.Add(facebookId, imageBitmap);
+
+            // Also add it to the list of recently accessed
+            List<long> listRecentlyAccessed = GetObjectField<List<long>>(userFbPicCache, "listUserFacebookIdRecentlyAccessed");
+            listRecentlyAccessed.Add(facebookId);
+
+            // Check what we can find from the cache
+            Bitmap bitmapFound = userFbPicCache.FindUserFacebookProfilePicture(facebookId);
+            Assert.True(bitmapFound != null);
+            Assert.True(bitmapFound.Equals(imageBitmap));
```

Comment:
It's bad practice to have a test case with more than one "assert", you might want to combine these into one "assert" statement. It helps pinpoint why a test case failed, and it follows the good quality assurance standards 👍 
Besides this the test cases look really good!